### ISSUE TITLE
Compile method Ignores Validation Errors.

### ISSUE THREAD

**Issue by ChandrashekharGanekal** [eclipse/xtext#1421]
Compile method in CompilationTestHelper class ignores the validation errors in DSL and continue with code generation, which is not correct.
Even though Result object has the validation error but ideally it should not generate the code also.

**Comment by ChandrashekharGanekal** [comment]
label:"good first issue"

**Comment by ChandrashekharGanekal** [comment]
This is my first issue, not able to add labels.
Please have a look.

**Comment by cdietrich** [comment]
If this is fixed I would prefer a solution that does not break existing tests.
@szarnekow what do you think

**Comment by ChandrashekharGanekal** [comment]
@cdietrich  thank you :)

**Comment by LorenzoBettini** [comment]
If I remember correctly the current behavior is intentional. This allows you to test your code generator in the presence of an invalid program. The presence of validation errors can be manually checked before generating code. https://github.com/szarnekow

**Comment by cdietrich** [comment]
Yes thus enhancement and wish for new api

**Comment by szarnekow** [comment]
True, the behavior as is was intentional. Some compilers may choose to gracefully deal with errors in the model.
